app/src/main/java/com/example/atheneum/fragments/newRequest.java

```diff
@@ -57,23 +88,52 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
 
         }
 
-        bookIDTest = this.view.findViewById(R.id.bookIDTest);
+        db = FirebaseDatabase.getInstance();
```

Comment:
It's not strictly necessary in order to implement the feature but it would be cleaner if you create a ViewModel that returns a LiveData<List<Book>> which you can observe and then make changes to your array adapter based on that. It's not in general a good idea to keep too much database logic in your activity since it makes the activity code very large and harder to read. In addition, there's nothing currently stopping you from making a request on your own book which doesn't really make sense. I think that you should only display books that you don't own in the book list.

Reply:
But that can also be done for part 5 and isn't strictly necessary for the part 4 deadline since I would rather have this feature done than be perfect